packages/1155-contracts/package/premint-api.test.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  createTestClient,
+  http,
+  createWalletClient,
+  createPublicClient,
+} from "viem";
+import { foundry, zoraTestnet } from "viem/chains";
+import { describe, it, beforeEach, expect, vi } from "vitest";
+import { parseEther } from "viem";
+import {
+  zoraCreator1155PremintExecutorImplABI as preminterAbi,
+  zoraCreator1155PremintExecutorImplAddress as zoraCreator1155PremintExecutorAddress,
+  zoraCreator1155ImplABI,
+  zoraCreator1155FactoryImplAddress,
+  zoraCreator1155FactoryImplConfig,
+} from "./wagmiGenerated";
+import ZoraCreator1155Attribution from "../out/ZoraCreator1155Attribution.sol/ZoraCreator1155Attribution.json";
+import zoraCreator1155PremintExecutor from "../out/ZoraCreator1155PremintExecutorImpl.sol/ZoraCreator1155PremintExecutorImpl.json";
+import zoraCreator1155Impl from "../out/ZoraCreator1155Impl.sol/ZoraCreator1155Impl.json";
+import zoraCreator1155FactoryImpl from "../out/ZoraCreator1155FactoryImpl.sol/ZoraCreator1155FactoryImpl.json";
+import zoraCreatorFixedPriceSaleStrategy from "../out/ZoraCreatorFixedPriceSaleStrategy.sol/ZoraCreatorFixedPriceSaleStrategy.json";
+import protocolRewards from "../out/ProtocolRewards.sol/ProtocolRewards.json";
+import {
+  ContractCreationConfig,
+  PremintConfig,
+  TokenCreationConfig,
+  preminterTypedDataDefinition,
+} from "./preminter";
+import { PreminterAPI } from "./premint-api";
+
+const walletClient = createWalletClient({
+  chain: foundry,
+  transport: http(),
+});
+
+export const walletClientWithAccount = createWalletClient({
+  chain: foundry,
+  transport: http(),
+});
+
+const testClient = createTestClient({
+  chain: foundry,
+  mode: "anvil",
+  transport: http(),
+});
+
+const publicClient = createPublicClient({
+  chain: foundry,
+  transport: http(),
+});
+
+type Address = `0x${string}`;
+
+const zeroAddress: Address = "0x0000000000000000000000000000000000000000";
+
+// JSON-RPC Account
+const [deployerAccount, creatorAccount, collectorAccount] =
+  (await walletClient.getAddresses()) as [Address, Address, Address, Address];
+
+type TestContext = {
+  preminterAddress: `0x${string}`;
+  forkedChainId: keyof typeof zoraCreator1155FactoryImplAddress;
+  anvilChainId: number;
+  zoraMintFee: bigint;
+  fixedPriceMinterAddress: Address;
+};
+
+describe("ZoraCreator1155Preminter", () => {
+  beforeEach<TestContext>(async (ctx) => {
+    // deploy signature minter contract
+    await testClient.setBalance({
```

Comment:
id remove this here, and put this down below in the test to right before where the eth is needed, and load up the account with the exact amount that is needed